src/io/network/endpoint.hpp

```diff
@@ -33,7 +33,7 @@ struct Endpoint {
   Endpoint &operator=(Endpoint &&) noexcept = default;
   ~Endpoint() = default;
 
-  enum class IpFamily : std::uint8_t { NONE, IP4, IP6 };
+  enum class IpFamily : std::uint8_t { NONE, IP4, IP6, DNS_ALIAS };
```

Comment:
DNS_ALIAS does not make sense for IpFamily

Reply:
Then when we register a replica with url instead of ip address should we just skip this IpFamily check? I'm asking this because when Endpoint is created it will throw an error if the address isn't neither IP4 nor IP6.